### PR TITLE
[FIX] ActionButton: update SVG icon to display current style in top bar

### PR DESCRIPTION
**Pull request by aath-odoo**
## Description:

Upadated the SVG icon in the top bar to reflect the current style. The icon now correctly displays the appropriate style based on the selected style.

Odoo task ID : [3341969](https://www.odoo.com/web#id=3341969&cids=2&menu_id=3940&action=4043&model=project.task&view_type=form)

## review checklist

- [ ] feature is organized in plugin, or UI components
- [ ] support of duplicate sheet (deep copy)
- [ ] in model/core: ranges are Range object, and can be adapted (adaptRanges)
- [ ] in model/UI: ranges are strings (to show the user)
- [ ] undo-able commands (uses this.history.update)
- [ ] multiuser-able commands (has inverse commands and transformations where needed)
- [ ] new/updated/removed commands are documented
- [ ] exportable in excel
- [ ] translations (\_lt("qmsdf %s", abc))
- [ ] unit tested
- [ ] clean commented code
- [ ] track breaking changes
- [ ] doc is rebuild (npm run doc)
- [ ] status is correct in Odoo

### CODE REVIEW

File: tests/components/top_bar.test.ts [hunk 1]
```diff
@@ -342,6 +356,21 @@ describe("TopBar component", () => {
       await click(fixture, `.o-menu-item-button[title="${iconTitle}"]`);
       expect(model.getters.getCurrentStyle()).toEqual(expectedStyle);
     });
+    test.each([
+      ["Top", { verticalAlign: "top" }],
+      ["Middle", { verticalAlign: "middle" }],
+      ["Bottom", { verticalAlign: "bottom" }],
+    ])("can set vertical alignmen '%s't with the toolbar", async (iconTitle, expectedStyle) => {
```
Comment (rrahir) [inline tests/components/top_bar.test.ts:363]:
```suggestion
    ])("can set vertical alignment '%s' with the toolbar", async (iconTitle, expectedStyle) => {
```

File: tests/components/top_bar.test.ts [hunk 2]
```diff
@@ -309,6 +309,20 @@ describe("TopBar component", () => {
       await click(fixture, `.o-menu-item-button[title="${iconTitle}"]`);
       expect(model.getters.getCurrentStyle()).toEqual(expectedStyle);
     });
+    test.each([["Left (Ctrl+Shift+L)"], ["Center (Ctrl+Shift+E)"], ["Right (Ctrl+Shift+R)"]])(
+      "Update icon on change",
+      async (iconTitle) => {
```
Comment (rrahir) [inline tests/components/top_bar.test.ts:314]: You should directly test against the actions and not look at how they are rendered. I suggest you look at how Lucas did it here : https://github.com/odoo/o-spreadsheet/pull/2651/files#diff-e729fc703acf88b1fee2adb7d8a03b2081192d64b7ae73e52dc7cee34fc7d306R879  He's testing the dynamic change of the key `description` so it's really close to what you are doing :)

File: tests/components/top_bar.test.ts (hunk 1 above)
Comment (rrahir) [reply tests/components/top_bar.test.ts:363]: same comment as above

File: tests/components/top_bar.test.ts [hunk 3]
```diff
@@ -375,6 +404,20 @@ describe("TopBar component", () => {
       await click(fixture, `.o-menu-item-button[title="${iconTitle}"]`);
       expect(model.getters.getCurrentStyle()).toEqual(expectedStyle);
     });
+    test.each([["Overflow"], ["Wrap"], ["Clip"]])(
+      "can set the wrapping state '%s' with the toolbar",
```
Comment (rrahir) [inline tests/components/top_bar.test.ts:408]: same comment as above

**Review by rrahir — CHANGES_REQUESTED** [review]

File: tests/components/top_bar.test.ts (hunk 2 above)
Comment (aath-odoo) [reply tests/components/top_bar.test.ts]: Hello @rrahir ,I attempted the approach you recommended. However, due to the absence of a registry for action buttons on the toolbar, I couldn't utilize the getNode() as it uses topbarmenuregistry function as shown in Lucas's pull request. Nevertheless, I devised an alternative method to associate the icon with the current cell style. Please take a look.

File: tests/components/top_bar.test.ts [hunk 4]
```diff
@@ -873,3 +873,25 @@ test("The menu items are orderer by their sequence", async () => {
...
+  [{ align: "right" }, "align-right", "Horizontal align"],
+  [{ wrapping: "clip" }, "wrapping-clip", "Wrapping"],
+  [{ wrapping: "wrap" }, "wrapping-wrap", "Wrapping"],
+  [{ wrapping: "overflow" }, "wrapping-overflow", "Wrapping"],
+])(
+  "Icon in top bar matches the selected cell style",
+  async (style, expectedTitleActive, buttonTitle) => {
+    const model = new Model();
+    setStyle(model, "A1", style as Style);
+    await mountParent(model);
```
Comment (rrahir) [inline tests/components/top_bar.test.ts:892]: wrap this test in a describe where you instantiate a single parent then in the test.each, set the style, await a next tick

**Review by rrahir — CHANGES_REQUESTED** [review]
The implementation became way to generic and tries to handle apples an oranges all the same. You should keep the distinction between the different kind of options (wrapping vs horizontal vs vertical).
 I think the implementation of 2-3 weeks ago was better, except for the helpers that you could add to improve.


I would like @somu-odoo  to go through the pr as well onece you do the changes :)

File: src/helpers/ui/icon_alignment.ts [hunk 5]
```diff
@@ -0,0 +1,64 @@
+import {
+  DEFAULT_HORIZONTAL_ALIGN,
+  DEFAULT_VERTICAL_ALIGN,
+  DEFAULT_WRAPPING_MODE,
+} from "../../constants";
+import { SpreadsheetChildEnv } from "../../types";
+
+export function getIconAlignment(env: SpreadsheetChildEnv, alignmentType: string) {
```
[the remark this answers is not in the document]
Comment (aath-odoo) [reply src/helpers/ui/icon_alignment.ts]: Okay i've done the changes and got the previous implementation of the fix if it's what you were referring to. It seems you're right we don't need to make it too generic, helpers functions did the job :), although do you think we should update the icons in topbar menu as well or we can keep them as is?

[the remark this answers is not in the document]
Comment (rrahir) [reply src/helpers/ui/icon_alignment.ts]: I think it's good as it is :)

**Comment by rrahir** [comment]
@robodoo r+

**Pull request closed without merging**